src/network/connection.cpp

```diff
@@ -0,0 +1,594 @@
+
+	MUST(sendByte(PACKET_INSPECTION_PACKETTYPE_RAW_IMU_DATA));
+
+	MUST(sendByte(sensorId));
+	MUST(sendByte(PACKET_INSPECTION_DATATYPE_FLOAT));
+
+	MUST(sendFloat(rX));
+	MUST(sendFloat(rY));
+	MUST(sendFloat(rZ));
+	MUST(sendByte(rA));
+
+	MUST(sendFloat(aX));
+	MUST(sendFloat(aY));
+	MUST(sendFloat(aZ));
+	MUST(sendByte(aA));
+
+	MUST(sendFloat(mX));
+	MUST(sendFloat(mY));
+	MUST(sendFloat(mZ));
+	MUST(sendByte(mA));
+
+	MUST(endPacket());
+}
+#endif
+
+void Connection::returnLastPacket(int len) {
+	MUST(m_Connected);
+
+	MUST(beginPacket());
+
+	MUST(sendBytes(m_Packet, len));
+
+	MUST(endPacket());
+}
+
+void Connection::updateSensorState(Sensor* const sensor1, Sensor* const sensor2) {
+	if (millis() - m_LastSensorInfoPacketTimestamp <= 1000) {
+		return;
+	}
+
+	m_LastSensorInfoPacketTimestamp = millis();
+
+	if (m_AckedSensorState1 != sensor1->getSensorState()) {
+		sendSensorInfo(sensor1);
+	}
+
+	if (m_AckedSensorState2 != sensor2->getSensorState()) {
+		sendSensorInfo(sensor2);
+	}
+}
+
+void Connection::searchForServer() {
+	while (true) {
+		int packetSize = m_UDP.parsePacket();
+		if (!packetSize) {
+			break;
+		}
+
+		// receive incoming UDP packets
+		int len = m_UDP.read(m_Packet, sizeof(m_Packet));
+
+#ifdef DEBUG_NETWORK
+		m_Logger.trace(
+			"Received %d bytes from %s, port %d",
+			packetSize,
+			m_UDP.remoteIP().toString().c_str(),
+			m_UDP.remotePort()
+		);
+		m_Logger.traceArray("UDP packet contents: ", m_Packet, len);
+#endif
+
+		// Handshake is different, it has 3 in the first byte, not the 4th, and data
+		// starts right after
+		if (m_Packet[0] == PACKET_HANDSHAKE) {
+			if (strncmp((char*)m_Packet + 1, "Hey OVR =D 5", 12) != 0) {
+				m_Logger.error("Received invalid handshake packet");
+				continue;
+			}
+
+			m_ServerHost = m_UDP.remoteIP();
+			m_ServerPort = m_UDP.remotePort();
+			m_LastPacketTimestamp = millis();
+			m_Connected = true;
+
+			statusManager.setStatus(SlimeVR::Status::SERVER_CONNECTING, false);
+			ledManager.off();
+
+			m_Logger.debug(
+				"Handshake successful, server is %s:%d",
+				m_UDP.remoteIP().toString().c_str(),
+				m_UDP.remotePort()
+			);
+
+			break;
+		}
+	}
+
+	auto now = millis();
+
+	// This makes the LED blink for 20ms every second
+	if (m_lastConnectionAttemptTimestamp + 1000 < now) {
+		m_lastConnectionAttemptTimestamp = now;
+		m_Logger.info("Searching for the server on the local network...");
+		Connection::sendTrackerDiscovery();
+		ledManager.on();
+	} else if (m_lastConnectionAttemptTimestamp + 20 < now) {
+		ledManager.off();
+	}
+}
+
+void Connection::reset() {
+	m_Connected = false;
+	m_AckedSensorState1 = SensorStatus::SENSOR_OFFLINE;
+	m_AckedSensorState2 = SensorStatus::SENSOR_OFFLINE;
+
+	m_UDP.begin(m_ServerPort);
+
+	statusManager.setStatus(SlimeVR::Status::SERVER_CONNECTING, true);
+}
+
+void Connection::update() {
+	auto sensor1 = sensorManager.getFirst();
+	auto sensor2 = sensorManager.getSecond();
+
+	if (!m_Connected) {
+		searchForServer();
```

Comment:
Add a return here to cut short when client is not connected. Otherwise, the logger on timeout detection would spam warnings as no new packets to update `m_LastPacketTimestamp`.
```suggestion
		searchForServer();
		return;
```